### PR TITLE
Catch up with Core until commit f617e05c38c9c7e663b824fefe5329cc5cbbf26a

### LINKED ISSUE #532: move bitcoin compatibility tests back to regtest
Instead of regtest2, which is a custom chain in `test/bitcoin_functional/functional` tests.

This would make syncing upstream less painful for these tests, but would also mean less exercising of our various bitcoin compatibility flags.

### PR DESCRIPTION
**Pull request by stevenroose**
Currently this catches up until Core commit f617e05c38c9c7e663b824fefe5329cc5cbbf26a and has everything working except the `p2p_invalid_messages` unit test that gets some updates in later upstream commits.

I'm planning to continue to catch up to 519b0bc5dc5155b6f7e2362c2105552bb7618ad0 next, which is the merge-base between upstream master and upstream v0.18.0.

Fixes #532.

### CODE REVIEW

**Comment by stevenroose** [comment]
@instagibbs If it's ok for you, I'm gonna take out those two commits without `MERGE-FIX` and PR them separately so they can be merged before this and I can take them out.

**Comment by instagibbs** [comment]
Yes please do, I've ACK'd them

**Comment by instagibbs** [comment]
linter:
```
Duplicate include(s) in src/rpc/rawtransaction.cpp:
#include <keystore.h>
#include <rpc/util.h>
```

**Comment by instagibbs** [comment]
```
test/functional/test_framework/liquid_addr.py:16: invalid syntax at "chk = ((chk & 0x7fffffffffffff) << 5) ^^ value # 0x1ffffff->0x7fffffffffffff"
test/functional/feature_bip68_sequence.py:33: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/feature_confidential_transactions.py:45: unused attribute 'is_network_split' (60% confidence)
test/functional/feature_default_asset_name.py:29: unused attribute 'is_network_split' (60% confidence)
test/functional/feature_fedpeg.py:181: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/feature_initial_reissuance_token.py:32: unused attribute 'is_network_split' (60% confidence)
test/functional/feature_issuance.py:104: unused attribute 'is_network_split' (60% confidence)
test/functional/feature_mandatory_coinbase.py:17: unused variable 'mandatory_pubkey' (60% confidence)
test/functional/feature_nulldummy.py:51: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/feature_pak.py:68: unused variable 'validate' (100% confidence)
test/functional/feature_progress.py:31: unused attribute 'is_network_split' (60% confidence)
test/functional/feature_txwitness.py:217: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/p2p_compactblocks.py:798: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/p2p_segwit.py:224: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/rpc_calcfastmerkleroot.py:13: unused attribute 'node_fastmerkle' (60% confidence)
test/functional/test_framework/messages.py:417: unused function 'setToAsset' (60% confidence)
test/functional/wallet_bumpfee.py:41: unused attribute 'node_fastmerkle' (60% confidence)
^---- failure generated from test/lint/lint-python-dead-code.sh
```

File: test/bitcoin_functional/functional/test_framework/util.py [hunk 1]
```diff
@@ -302,6 +302,26 @@ def initialize_datadir(dirname, n):
         f.write("discover=0\n")
         f.write("listenonion=0\n")
         f.write("printtoconsole=0\n")
+        # Elements:
```
Comment (instagibbs) [inline test/bitcoin_functional/functional/test_framework/util.py]: If we're using regtest I don't think these are required... I can do a follow-up PR if true

File: test/functional/feature_pak.py [hunk 2]
```diff
@@ -329,9 +334,6 @@ def compare(actual, expected):
 
         assert_raises_rpc_error(-8, "bitcoin_descriptor must be a ranged descriptor.", self.nodes[i_pak1].initpegoutwallet, "pkh(tpubD6NzVbkrYhZ4WaWSyoBvQwbpLkojyoTZPRsgXELWz3Popb3qkjcJyJUGLnL4qHHoQvao8ESaAstxYSnhyswJ76uZPStJRJCTKvosUCJZL5B)")
 
-        # key origins aren't supported in 0.17
```
Comment (instagibbs) [inline test/functional/feature_pak.py]: nit: we should add "positive" tests since they're supported now... not sure what it'll do

**Comment by instagibbs** [comment]
note that this PR resolves https://github.com/ElementsProject/elements/issues/532

File: test/bitcoin_functional/functional/test_framework/util.py (hunk 1 above)
Comment (stevenroose) [reply test/bitcoin_functional/functional/test_framework/util.py]: none of them?

Comment (instagibbs) [reply test/bitcoin_functional/functional/test_framework/util.py]: I'll test later :)

**Comment by instagibbs** [comment]
Last clang issues:
```
test/mempool_tests.cpp:448:19: warning: unused variable 'tx1Hash' [-Wunused-variable]
    const uint256 tx1Hash(tx.GetHash());
                  ^
1 warning generated.
```
and
```
wallet/rpcwallet.cpp:1201:17: warning: unused function 'addwitnessaddress' [-Wunused-function]
static UniValue addwitnessaddress(const JSONRPCRequest& request)
                ^
1 warning generated.
```
and
```
wallet/wallet.cpp:2692:59: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
        std::map<uint256, CWalletTx>::const_iterator it = mapWallet.find(presetInput.hash);
                                                          ^
wallet/wallet.cpp:2693:19: warning: reading variable 'mapWallet' requires holding mutex 'cs_wallet' [-Wthread-safety-analysis]
        if (it != mapWallet.end()) {
                  ^
2 warnings generated.
```

**Comment by instagibbs** [comment]
utACK https://github.com/ElementsProject/elements/pull/620/commits/693f206082aa3b0e9b97de6ee5f9f3d42486047b

**Pull request merged**